docarray/array/storage/weaviate/backend.py

```diff
@@ -234,14 +327,10 @@ def _load_or_create_weaviate_schema(self):
         :return: the schemas of this :class`DocumentArrayWeaviate` object and its meta
         """
         if not self._config.name:
-            name_candidate = self._get_weaviate_class_name()
-            doc_schemas = self._get_schema_by_name(name_candidate)
-            while self._client.schema.contains(doc_schemas):
-                name_candidate = self._get_weaviate_class_name()
-                doc_schemas = self._get_schema_by_name(name_candidate)
-            self._client.schema.create(doc_schemas)
-            self._config.name = name_candidate
-            return doc_schemas
+            raise ValueError(
```

Comment:
this is too much of a breaking change. It is a wanted feature to have a default name

Reply:
This is the Weaviate client configuration when choosing Weaviate as a storage engine (hence it's in `_init_storage()`)

I've added additional information in the code itself

Reply:
Very good point, this is changed